manifests/scripts/metrics-dashboard/run.sh

```diff
@@ -0,0 +1,109 @@
+
+echo
+echo "* Waiting for Grafana CRDs to exist"
+
+while : ; do
+  kubectl get customresourcedefinition/grafanas.integreatly.org  > /dev/null 2>&1 && break
+  sleep 1s
+done
+
+# -----------------
+
+echo "* Waiting for Grafana route"
+
+while : ; do
+  kubectl get route/grafana-route -n grafana -o yaml  > /dev/null 2>&1 && break
+  sleep 1s
+done
+
+export HOSTNAME=`kubectl get route/grafana-route -n grafana -o yaml  | grep "    host:" | cut -c11-`
+export ADMIN_SECRET_VALUE=`openssl rand -hex 10`
+
+echo
+echo "* Grafana route is: https://$HOSTNAME"
+echo "  Username: user"
+echo "  Password: $ADMIN_SECRET_VALUE"
+echo
+
+TMP_DIR=`mktemp -d`
+
+# Substitute the cluster domain into the Grafana Ingress CR
+cp -f grafana/grafana-cr.yaml $TMP_DIR/grafana-cr-resolved.yaml
+sed -i 's/HOSTNAME/'$HOSTNAME'/g' $TMP_DIR/grafana-cr-resolved.yaml
+sed -i 's/ADMIN_SECRET_VALUE/'$ADMIN_SECRET_VALUE'/g' $TMP_DIR/grafana-cr-resolved.yaml
+
+
+kubectl apply -f $TMP_DIR/grafana-cr-resolved.yaml -n grafana
+
+rm -f "$TMP_DIR/grafana-cr-resolved.yaml"
+
+# The kubectl equivalent to: 'oc adm policy add-cluster-role-to-user cluster-monitoring-view -z grafana-serviceaccount'
+kubectl apply -f grafana/grafana-cluster-role-binding.yaml -n grafana
+
+# -----------------
+echo
+echo "* Waiting for Grafana service account token secret to exist"
+while : ; do
+  kubectl get secrets -n grafana | grep "grafana-serviceaccount-token"  > /dev/null 2>&1 && break
+  sleep 1s
+done
+
+echo
+echo "* Applying GrafanaDataSource, using Grafana Service Account Token"
+
+GRAFANA_SECRET=`kubectl -n grafana get secrets | grep "grafana-serviceaccount-token" |  cut -d ' ' -f 1`
+
+GRAFANA_SA_TOKEN=`kubectl -n grafana get secret $GRAFANA_SECRET -o jsonpath={.data.token} | base64 -d`
+
+cp -f grafana/grafana-data-source.yaml  $TMP_DIR/grafana-data-source-resolved.yaml
+
+sed -i 's/GRAFANA_SA_TOKEN/'$GRAFANA_SA_TOKEN'/g' $TMP_DIR/grafana-data-source-resolved.yaml
+
+kubectl apply -f $TMP_DIR/grafana-data-source-resolved.yaml
+
+rm -f "$TMP_DIR/grafana-data-source-resolved.yaml"
+
+# This section was based on https://www.redhat.com/en/blog/custom-grafana-dashboards-red-hat-openshift-container-platform-4
+
+# -----------------
+echo
+echo "* Create Argo CD dashboards"
+
+# Argo CD dashboard is based on example Grafana dashboard from upstream Argo CD docs
+kubectl apply -f dashboards/argo-cd/argo-grafana-dashboard-cm.yaml
+kubectl apply -f dashboards/argo-cd/grafana-argo-dashboard.yaml
+
+# Create a ServiceMonitor for GitOps Operator, in the openshift-gitops namespace
+# - This SHOULD instead be created in the openshift-operators namespace (which is where the gitops-operator lives),
+#   but the prometheus-operator process has a hardcoded list of namespaces that it checks, openshift-operators
+#   is not on it.
+
+echo "* Create GitOps Operator ServiceMonitor"
+
+kubectl apply -f prometheus/openshift-operators-service-monitor.yaml -n openshift-gitops
```

Comment:
Aren't we supposed to set up monitoring for the gitops-service-operator rather than the gitops-operator?

Reply:
I've updated the README.md with more context on what is being setup.